server: scale DPI with signed arithmetic so negative coordinates survive.

--- a/server/user.h
+++ b/server/user.h
@@ -234,9 +234,9 @@
 
 static inline int scale_dpi( int val, struct ratio dpi_from, struct ratio dpi_to )
 {
-    unsigned int num = dpi_to.num * dpi_from.den, den = dpi_from.num * dpi_to.den;
-    if (val >= 0) return (val * num + (den / 2)) / den;
-    return (val * num - (den / 2)) / den;
+    int num = dpi_to.num * dpi_from.den, den = dpi_from.num * dpi_to.den;
+    if (val >= 0) return ((long long)val * num + (den / 2)) / den;
+    return ((long long)val * num - (den / 2)) / den;
 }
 
 static inline void scale_dpi_rect( struct rectangle *rect, struct ratio dpi_from, struct ratio dpi_to )
